tests/decider_tests.py

```diff
@@ -0,0 +1,289 @@
+            f.seek(0)
+            yield f
+
+    def test_get_variant_expose_event_fields(self):
+        config = {
+            "exp_1": {
+                "id": 1,
+                "name": "exp_1",
+                "enabled": True,
+                "version": "2",
+                "type": "range_variant",
+                "start_ts": 37173982,
+                "stop_ts": 2147483648,
+                "owner": "test_owner",
+                "experiment": {
+                    "variants": [
+                        {"range_start": 0.0, "range_end": 0.2, "name": "control_1"},
+                        {"range_start": 0.2, "range_end": 0.4, "name": "control_2"},
+                        {"range_start": 0.4, "range_end": 0.6, "name": "variant_2"},
+                        {"range_start": 0.6, "range_end": 0.8, "name": "variant_3"},
+                        {"range_start": 0.8, "range_end": 1.0, "name": "variant_4"},
+                    ],
+                    "experiment_version": 2,
+                    "shuffle_version": 0,
+                    "bucket_val": "user_id",
+                    "log_bucketing": False,
+                },
+            }
+        }
+
+        with self.create_temp_config_file(config) as f:
+            filewatcher = FileWatcher(path=f.name, parser=init_decider_parser, timeout=2, backoff=2)
+            decider = Decider(
+                decider_context=self.minimal_decider_context,
+                config_watcher=filewatcher,
+                server_span=self.mock_span,
+                context_name="test",
+                event_logger=self.event_logger,
+            )
+
+            self.assertEqual(self.event_logger.log.call_count, 0)
+            variant = decider.get_variant("exp_1")
+            self.assertEqual(variant, "variant_4")
+
+            # Todo: enable expose call
+            # self.assertEqual(self.event_logger.log.call_count, 1)
+
+            # event_fields = self.event_logger.log.call_args[1]
+            # self.assertEqual(event_fields["variant"], "variant_4")
+            # self.assertEqual(event_fields["user_id"], "t2_1234")
+            # self.assertEqual(event_fields["logged_in"], True)
+            # self.assertEqual(event_fields["app_name"], None)
+            # self.assertEqual(event_fields["cookie_created_timestamp"], 10000)
+            # self.assertEqual(event_fields["event_type"], "expose")
+            # self.assertNotEqual(event_fields["span"], None)
+
+            # self.assertEqual(getattr(event_fields["experiment"], "id"), config["id"])
+            # self.assertEqual(getattr(event_fields["experiment"], "name"), config["name"])
+            # self.assertEqual(getattr(event_fields["experiment"], "owner"), config["owner"])
+            # self.assertEqual(getattr(event_fields["experiment"], "version"), config["experiment"]["experiment_version"])
+
+    def test_none_returned_on_variant_call_with_bad_id(self):
+        config = {
+            "test": {
+                "id": "1",
```

Comment:
What will be the case to cause the id to be passed as str?

Reply:
just some general error handling, nothing particular--I took this test case from [existing tests](https://github.com/reddit/experiments.py/blob/c5f8373d051845ab550a9ae65041afcc4c9f996b/tests/experiment_tests.py#L563-L566)

Reply:
I see. 
It is a non blocker but I think if ZK config is somehow passing `str` id, maybe we should not silently return None.

Reply:
It's definitely a serious issue if that happens but I don't think we should raise an exception..at least it seems that the mentality of the experiments.py sdk is to [log to warning](https://github.com/reddit/experiments.py/pull/2) instead of raising exceptions, which I do [here](https://github.com/reddit/experiments.py/pull/6/files#diff-00f6e0196121fdac7fda7102cc51166ba1e97b773541f1b7788f34dd6637c3f9R139) (rust lib provides specific failure message to std out as well):

```
err: SerdeError(
    Error("invalid type: string \"1\", expected u32", line: 1, column: 19),
)
WARNING  reddit_decider:__init__.py:139 Rust decider did not initialize.
```

Reply:
maybe `Rust decider did not initialize.` should be `error` level log, instead of warning though?

Reply:
I made the `x = rust_decider.init()` return value have a way to access the error string via `x.err()` (in [this pr](https://github.snooguts.net/reddit/decider/pull/16)) so that we can control the error message in the sdk and log it to logger.error() without having rust lib print to std out and pollute logs.